lib/ancestry/instance_methods.rb

```diff
@@ -151,17 +157,45 @@ def ancestor_of?(node)
     # assuming that parent hasn't changed
     def parent= parent
       self.ancestor_ids = parent ? parent.path_ids : []
+      clear_memoized_instance_variables
     end
 
     def parent_id= new_parent_id
       self.parent = new_parent_id.present? ? unscoped_find(new_parent_id) : nil
+      clear_memoized_instance_variables
     end
 
     def parent_id
-      ancestor_ids.last if ancestors?
+      return @_parent_id if defined?(@_parent_id)
+      @_parent_id = if @_ancestor_ids
+        @_ancestor_ids.empty? ? nil : @_ancestor_ids.last
```

Comment:
Oh, you just really wanted to trigger my OCD (not real) with this indenting here, huh?